core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/model/CloseableIterator.java

```diff
@@ -0,0 +1,17 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eclipse RDF4J contributors.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Distribution License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/org/documents/edl-v10.php.
+ ******************************************************************************/
+
+package org.eclipse.rdf4j.sail.memory.model;
+
+import java.util.Iterator;
+
+public interface CloseableIterator<E> extends Iterator<E>, AutoCloseable {
```

Comment:
There is a `AbstractCloseableIterator` in `org.eclipse.rdf4j.common.iterator`. It's currently not `AutoCloseable`, just `Closeable`, but that should be simple to fix. Can you reuse that - or alternative put this interface in that package and make `AbstractCloseableIterator` extend it?

Reply:
I moved the interface and renamed it. It's not possible to integrate it with the AbstractCloseableIterator because that one specifies that close should throw an IOException.

Reply:
I’m wondering if maybe I can just remove the entire interface now that I’m using a class instead of a lambda.